src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -564,7 +565,8 @@ public LeafRope makeValidLeafRopeAsciiCompat(byte[] bytes, Encoding encoding, Co
                 int delta = StringSupport.encFastMBCLen(bytes, p, e, encoding);
 
                 if (delta < 0) {
-                    throw new UnsupportedOperationException("Code range is reported as valid, but is invalid for the given encoding: " + encoding.toString());
+                    errorProfile.enter();
+                    throw new UnsupportedOperationException("Code range is reported as valid, but is invalid for the given encoding: " + encodingName(encoding));
```

Comment:
Are you sure about that? We have plenty of places where we concatenate strings not behind a boundary. `String.format` is problematic though.